### PR TITLE
[Wake] Feat: add partner token cookie and use in every request that suports It

### PR DESCRIPTION
**Pull request by IncognitaDev**
## What is this Contribution About?

This PR adds function to set and delete partner token cookie and use in every request that suports this token.
This PR also change the partner prop of wake ProductListingPage to make this prop more meaningful

### CODE REVIEW

File: wake/loaders/productDetailsPage.ts [hunk 1]
```diff
@@ -31,8 +32,12 @@ async function loader(
   const { slug, buyTogether, includeSameParent } = props;
   const { storefront } = ctx;
 
+  const partnerAccessToken = getPartnerCookie(req.headers);
+
   const headers = parseHeaders(req.headers);
 
+  console.log({ partnerAccessToken });
```
Comment (guitavano) [inline wake/loaders/productDetailsPage.ts:39]: remove

File: shopify/utils/storefront/storefront.graphql.gen.ts [hunk 2]
```diff
@@ -7704,6 +7704,8 @@ export type FilterFragment = { id: string, label: string, type: FilterType, valu
...
 
+export type CustomerFragment = { id: string, email?: string | null, firstName?: string | null, lastName?: string | null };
+
```
Comment (guitavano) [inline shopify/utils/storefront/storefront.graphql.gen.ts:7708]: Should be here? shopify folder?

File: wake/loaders/productListingPage.ts [hunk 3]
```diff
@@ -107,9 +108,10 @@ export interface Props {
   pageHref?: string;
 
   /**
+   * @title Partner Param
    * @description page param to partners page
    */
-  slug?: RequestURLParam;
+  partnerAlias?: RequestURLParam;
```
Comment (guitavano) [inline wake/loaders/productListingPage.ts:114]: It brokes the current sites using wake api, right?

**Review by guitavano — CHANGES_REQUESTED** [review]

File: shopify/utils/storefront/storefront.graphql.gen.ts (hunk 2 above)
Comment (IncognitaDev) [reply shopify/utils/storefront/storefront.graphql.gen.ts:7708]: Maybe the rebase make this, i will remove this

File: wake/loaders/productListingPage.ts (hunk 3 above)
Comment (matheusgr) [reply wake/loaders/productListingPage.ts:114]: we may have other clients that may need it.

Can you keep the slug property, tag it as deprecated, and define parterAlias with slug if it is not defined (i.e. `partnerAlias ??= slug`)?

**Review by matheusgr — APPROVED** [review]

**Pull request merged**